.travis.yml

```diff
@@ -18,13 +18,17 @@ install:
   # Install dependencies
   - conda create -n test-environment python=$TRAVIS_PYTHON_VERSION
   - source activate test-environment
-  - conda install pytest numpy pip coverage toolz pandas cython scikit-learn cytoolz dill networkx pyzmq psutil
+  - conda install pytest numpy pip coverage toolz pandas cython scikit-learn cytoolz dill networkx pyzmq psutil ipython
   - pip install chest
   - pip install git+https://github.com/blosc/bcolz --upgrade
 
   # Install dask
   - python setup.py install
 
+  # start the ipcluster, it takes a while to start
+  - ipcluster start -n 4 &
+  - sleep 20
```

Comment:
You found it. The setup/teadown are [here](https://github.com/ipython/ipyparallel/blob/master/ipyparallel/tests/__init__.py#L47), which includes waiting for engines to start.

I'm not happy with ipcluster at all, but the gist of waiting for engines is to connection a Client and wait for engines to register.